src/PowerShellEditorServices/Services/DebugAdapter/Handlers/ConfigurationDoneHandler.cs

```diff
@@ -43,10 +43,22 @@ internal class ConfigurationDoneHandler : IConfigurationDoneHandler
             _workspaceService = workspaceService;
         }
 
-        public Task<ConfigurationDoneResponse> Handle(ConfigurationDoneArguments request, CancellationToken cancellationToken)
+        public async Task<ConfigurationDoneResponse> Handle(ConfigurationDoneArguments request, CancellationToken cancellationToken)
         {
             _debugService.IsClientAttached = true;
 
+            if (_debugStateService.OwnsEditorSession)
+            {
+                // If this is a debug-only session, we need to start
+                // the command loop manually
+                _powerShellContextService.ConsoleReader.StartCommandLoop();
+                await Task.Delay(10000);
+                while (_powerShellContextService.SessionState != PowerShellContextState.Ready)
+                {
+                    await Task.Delay(2000);
+                }
+            }
+
```

Comment:
@SeeminglyScience

Reply:
`SessionState` should have a property-changed-like event. If `SessionState` starts at `Ready`, we should add another value to the enum `NotStarted` or something.  That would allow skipping the first delay.

Also maybe just have a `ManualResetEventSlim` that this response can `Wait` on.

Reply:
Maybe we put this to `NotStarted` https://github.com/PowerShell/PowerShellEditorServices/blob/master/src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs#L371